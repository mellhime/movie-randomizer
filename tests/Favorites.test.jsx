import {
  render,
  fireEvent,
  screen,
  act,
  renderHook,
} from "@testing-library/react";
import Favorites from "../src/scripts/Favorites";
import { useFetch } from "../src/scripts/useFetch";

describe("Favorites component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Favorites />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("increments count on button click", () => {
    render(<Favorites />);
    const button = screen.getAllByText(/Add to favorites/i)[0];
    act(() => {
      fireEvent.click(button);
    });
    expect(screen.getByTestId("count").textContent).toBe("Count: 1");
  });
});

describe("useFetch hook", () => {
  it("gets data", () => {
    const { result } = renderHook(() => useFetch("movies"));

    expect(result.current[0][0]).toHaveProperty("title");
  });
});
