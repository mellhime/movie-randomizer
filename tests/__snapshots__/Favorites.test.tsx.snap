// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component matches snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th>
          Название фильма
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Leon
        </td>
        <td>
          <button>
            Add to favorites
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Neon Genesis Evangelion
        </td>
        <td>
          <button>
            Add to favorites
          </button>
        </td>
      </tr>
      <tr>
        <td>
          2012
        </td>
        <td>
          <button>
            Add to favorites
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <p
    data-testid="count"
  >
    Count: 0
  </p>
</DocumentFragment>
`;
